dpctl-capi/source/vector_type.hpp

```diff
@@ -0,0 +1,36 @@
+//===-- vector_type.hpp - Define vector_class_t  ===//
+//
+//                      Data Parallel Control (dpctl)
+//
+// Copyright 2020-2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file is a private header of DPCTLSyclInterface to define
+/// vector_class_t, which used to corresponds to sycl::vector_class in earlier
+/// versions of DPC++ and later removed in favor of using std::vector
+///
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#if __SYCL_COMPILER_VERSION < 20210925
```

Comment:
For this preprocessor macro to be available, sycl.hpp needs to be included first.

Reply:
This was fixed, so resolving.